### PR TITLE
Option elements_container

### PR DESCRIPTION
**Pull request by SassNinja**
This PR adds the new option `elements_container` that only affects the queryOriginNode where the elements_selector is searched.

I'm doing some ajax content replacement where I only want to (re)initialize the replaced container instead of the whole document.
Currently the only way to adjust the queryOriginNode is to define the target container as `container` option but this does much more than only change the queryOriginNode.

I think the new option will be very useful for others who are in the same situation.

@verlok what do you think? Would you please review this and merge if it's ok for you?

I've created my feature branch from the development branch and thus I'm merging into that branch.
Just let me know if you prefer a PR into the master branch.

### CODE REVIEW

**Comment by verlok** [comment]
Hi, thank you for your service. 
I could merge this in, but can you explain better why can't you restrict the re-initialization using the `elements_selector` option and leaving `container` to your scrolling container?

**Comment by SassNinja** [comment]
@verlok I can't just use the `elements_selector` because it requires a string.
My reinitialization is node based (precisely jQuery objects). Thus I can't provide a selector string to the lazyload plugin.

my code looks similar to this
```javascript
window.reInit($container);
```
what reinitializes many different plugins in the target container. The way for me to do this is piping through the container to the lazyload plugin without affecting the container (window).

**Comment by verlok** [comment]
What if we made a modification to allow passing the elements directly to the LazyLoad object, like a `setElements` method? Would this solve?

**Comment by verlok** [comment]
Still, looking at the code you wrote,  it seems that you just need to change the queryOriginNode...  and I still don't understand why it cannot be the original container

**Comment by SassNinja** [comment]
> What if we made a modification to allow passing the elements directly to the LazyLoad object, like a `setElements` method? Would this solve?

It would be an improvement as well if the selector option also accepts elements but unfortunately this won't help me. I need an option to pass a container element in which the plugin is supposed to look for lazyload images and initializes them.

> Still, looking at the code you wrote, it seems that you just need to change the queryOriginNode... and I still don't understand why it cannot be the original container

Right I only need to change the queryOriginNode. Let me explain a bit more detailed why:

I'm working quite a lot with ajax on some pages and thus I need to reinitialize all the containers that I replace.
Since there are many plugins included (not only lazyload) I've created an abstract module where all reinits modules are bundled. To make things as easy as possible I've added a param to pipe though the element that got replaced. Then all registered modules look for the selectors (they internally use) within that element and initialize them.

Thus I would like to avoid getting away from my abstract setup and just need an option to tell lazyload to use a specific container instead of the whole document.
The current `container` option doesn't work because it's also used to determine when a lazyload image comes in. As said I only want to adjust the search scope.

Hope it's now clear why I need this option and why I've built it this way 😉

**Comment by verlok** [comment]
Great! It's clearer now. 
I think that if you use version 9 of LazyLoad that I released just a couple of days ago, you can now use the `container` option itself, since it's not used anymore to listen to the scroll event (version 9 uses the Intersection Observer API). 
Can you check this?

**Comment by verlok** [comment]
To be clearer, the `container` option is now used only as origin for the querySelectorAll method.

**Comment by verlok** [comment]
Never mind, I took a closer look to the code and I saw that it's also used as root element for the IntersectionObserver, so what I wrote before is not quite true.

**Comment by verlok** [comment]
I think that the best thing you can do is to create one instance of LazyLoad for each one of your modules. 

```js
var lazyLoadModuleN = new LazyLoad({
  elements_selector = "#moduleN .lazyElement"
});
```

Then when your module updates, you just do:

```js
lazyLoadModuleN.update();
```

This would ensure the best module separation. Right?

**Comment by SassNinja** [comment]
As you already spotted version 9 can't be used since it has the same limitation:
no option that only affects the search for elements

> I think that the best thing you can do is to create one instance of LazyLoad for each one of your modules.

I spent some time (before my PR) thinking about how to use the existing options to keep my modular reInit setup.
I also thought about the way you've suggested.

But the problem is I don't have a selector when the reInit is done.
Not all of the container that get reinitialzed have an id or only one class. So I can't provide a selector string if my whole setup is element based.

The bottom line is that there's simply an option missing that let's me pass in elements instead of a string.
Your previous suggestion to adjust the `elements_selector` option to accept elements as well is the only alternative to the PR imo.

But to be honest I don't understand what argues against my additional option.
It's just one additional option, a small condition at the top of the core code and provides 100% backwards compatibility.

What argues against it in your eyes?

**Comment by verlok** [comment]
I'm afraid that the users find the 2 options confusing, meaning `container` and `elements_comtainer`.

**Comment by verlok** [comment]
I mean, even if the options were called `scrolling_container` and `elements_container` it would be confusing. Why would the two be different? They are necessarily the same. 

That's why I asked you what you wanted to do and why couldn't you do it with the current options, and you answered "because `elements_selector` is a string (etc)". Thus, my logical solution is to make that option accept a set or dom elements. 

Is it clear now?

**Comment by SassNinja** [comment]
I indeed considered renaming `container` into `scrolling_container` because that option isn't described detailed enough in the docs (at the beginning I used it wrongly to adjust the queryOriginNode).

But I agree to you: even if the option gets renamed (what would break backwards compatibility) it might still be confusing.
So the best solution really seems to be extending the `elements_selector` option so it also supports elements.

As soon as I have some time I'll make another PR for this.

**Comment by verlok** [comment]
Thanks!
